tempodb/tempodb.go

```diff
@@ -544,3 +544,33 @@ func (rw *readerWriter) compactedBlocklist(tenantID string) []*encoding.Compacte
 
 	return copiedBlocklist
 }
+
+// updateBlocklist Add and remove regular or compacted blocks from the in-memory blocklist.
+// Changes are temporary and will be overwritten on the next poll.
+func (rw *readerWriter) updateBlocklist(tenantID string, add []*encoding.BlockMeta, remove []*encoding.BlockMeta, compactedAdd []*encoding.CompactedBlockMeta) {
+	if tenantID == "" {
+		return
+	}
+
+	rw.blockListsMtx.Lock()
+	defer rw.blockListsMtx.Unlock()
+
+	// ******** Regular blocks ********
+	toRemove := make(map[*encoding.BlockMeta]bool)
```

Comment:
nit:  map to empty struct

`map[*encoding.BlockMeta]struct{}`

Reply:
Done